### DISCUSSION TITLE
Flux Deployment Architecture/Models

### DISCUSSION THREAD

**Discussion by suseendare** [fluxcd/flux2#4456] [Q&A] [answered]
Hi Team,
Do we have recommendations/suggestions/architecture models on deploying Flux for production grade application?

For example, Argo has something - https://codefresh.io/blog/a-comprehensive-overview-of-argo-cd-architectures-2023/

**Comment by stefanprodan** [comment, marked as answer]
A single Flux instance with the default limits for controllers 1CPU/1Gi RAM can manager thousands apps with ease. We do have documentation on how to make Flux run at scale here:
- Vertical scaling https://fluxcd.io/flux/installation/configuration/vertical-scaling/
- Horizontal scaling https://fluxcd.io/flux/installation/configuration/sharding/
- Multi-tenancy https://fluxcd.io/flux/installation/configuration/multitenancy/

Starting with Flux 2.2.0 (scheduled for release next week) we publish benchmarks for running Flux at scale, see https://github.com/fluxcd/flux-benchmark

**Reply by suseendare** [reply]
Thanks Stefan

**Comment by stefanprodan** [comment]
I've published a comprehensive overview of Flux CD architectures for multi-cluster continuous deliver here https://control-plane.io/posts/fluxcd-architecture-overview/
